test/controllers/payment_authorization_controller_test.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:omise_flutter/src/controllers/payment_authorization_controller.dart';
+import 'package:omise_flutter/src/enums/enums.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+import '../mocks.dart';
+
+void main() {
+  late PaymentAuthorizationController controller;
+  late MockUriLauncher mockUriLauncher;
+
+  setUp(() {
+    mockUriLauncher = MockUriLauncher();
+    controller = PaymentAuthorizationController(
+        enableDebug: true, launchUrlFunction: mockUriLauncher.launchUrl);
+  });
+
+  group('PaymentAuthorizationController', () {
+    test('initial state is idle', () {
+      expect(controller.value.webViewLoadingStatus, Status.idle);
+      expect(controller.value.enableDebug, true);
+    });
+
+    test('updateState updates the controller state', () {
+      // New state to apply
+      final newState = PaymentAuthorizationState(
+        webViewLoadingStatus: Status.loading,
+        currentWebViewUrl: 'https://example.com',
+        enableDebug: true,
```

Comment:
I am just curious, why this test need to enableDebug? 🤔

Reply:
It doesn't matter, its is an internally required param and being false or true will not affect the result as this only logs some debug messages. I have set it as true to maximize the testability.